share/functions/__fish_man_page.fish

```diff
@@ -8,11 +8,21 @@ function __fish_man_page
         return
     end
 
+    #Skip `sudo`
+    set maincmdind 1
+    while set -q args[(math $maincmdind + 1)]
```

Comment:
How about
```fish
    while set -q args[(math $maincmdind + 1)]
        and test $args[$maincmdind] = "sudo"

        set maincmdind (math $maincmdind + 1)
    end
```

Or you could use the following and get away without using `math` at all
```fish
    while set -q args[2]
        and test $args[1] = "sudo"

        set args $args[2..-1]
    end
```

Reply:
> `set args $args[2..-1]`

aka `set -e foo[1]`